widget/button_internal_test.go

```diff
@@ -99,20 +99,14 @@ func TestButtonRenderer_Layout_Stretch(t *testing.T) {
 	button.Resize(button.MinSize().Add(fyne.NewSize(100, 100)))
 	render := test.WidgetRenderer(button).(*buttonRenderer)
 
-	iconYOffset, labelYOffset := 0, 0
 	textHeight := render.label.MinSize().Height
-	if theme.IconInlineSize() > textHeight {
-		labelYOffset = (theme.IconInlineSize() - textHeight) / 2
-	} else {
-		iconYOffset = (textHeight - theme.IconInlineSize()) / 2
-	}
 	minIconHeight := fyne.Max(theme.IconInlineSize(), textHeight)
 	assert.Equal(t, 50+theme.Padding()*3, render.icon.Position().X, "icon x")
-	assert.Equal(t, 50+theme.Padding()*2+iconYOffset, render.icon.Position().Y, "icon y")
+	assert.Equal(t, 50+theme.Padding()*2, render.icon.Position().Y, "icon y")
 	assert.Equal(t, theme.IconInlineSize(), render.icon.Size().Width, "icon width")
 	assert.Equal(t, minIconHeight, render.icon.Size().Height, "icon height")
 	assert.Equal(t, 50+theme.Padding()*4+theme.IconInlineSize(), render.label.Position().X, "label x")
-	assert.Equal(t, 50+theme.Padding()*2+labelYOffset, render.label.Position().Y, "label y")
+	assert.Equal(t, 50+theme.Padding()*2, render.label.Position().Y, "label y")
```

Comment:
I think this is neither, the tests were inconsistent with the code. In label the icon is not adjusted by changing it's position but by using the label resize to cause it's content to move (I think).
This change was required because the `0` int difference between them before was ignored, but now it is `0.5` which the actual layout code did not set, but the test expected.

Reply:
I just had a look at the button layout implementation. Your change is okay but I think the button layout needs an overhaul. It’s a mix of lay out by `Layout` and explicit moves and resizes. But, of course, this is off topic for this PR.